### ISSUE TITLE
[x86 release Jit32]  Branch into exception handler block doesn't result in InvalidProgramException but leading to an AV, since verification is turned OFF by CoreCLR VM

### ISSUE THREAD

**Issue by sivarv** [dotnet/coreclr#5286]
Tests that are failing:
JIT\Regression\Dev11\Dev11_468598\Test_HndIndex_10_Plain\Test_HndIndex_10_Plain.exe
JIT\Regression\Dev11\Dev11_468598\Test_HndIndex_10_Reordered\Test_HndIndex_10_Reordered.exe

These tests pass on dbg/chk builds because fgMakeBasicBlocks() after finding basic blocks, calls fgCheckBasicBlockControlFlow()  always on dbg/chk builds and in retail builds only when verification is enabled.  It is fgCheckBAsicBlockControlFlow() that detects branch into catch handler is invalid and throws InvalidProgramException.

```
#ifndef DEBUG
    if (tiVerificationNeeded)
#endif
    {
        // always run these checks for a debug build
        verCheckNestingLevel(initRoot);
        fgCheckBasicBlockControlFlow();
    }
```

The reason on release build fgCheckBasicBlockControlFlow() doesn't get called is that CoreCLR VM is passing CORJIT_FLG_SKIP_VERIFICATION flag in compileFlags that turns OFF verification and as a result fgCheckBasicBlockControlFlow() doesn't get executed and hence the failure.

I tried the same test case on Desktop .Net 4.6.x installed on my machine and the tests passed by detecting the expected InvalidProgramException.  The reason is that the test has  SecurityTransparentAttribute, that will trigger verification on Desktop.  Whereas CoreCLR creates full trust AppDomains and doesn't honor transparency attributes. If SecurityTransparentAttribute is deleted and the test is run on Desktop .Net 4.6.x, it too AVs since Verification will be OFF.

In summary, these tests AV'ing on release builds is expected on CoreCLR, since it no longer honors transparency attributes.  Filing this issue to disable these tests.

**Comment by pgavlin** [comment]
@RussKeldorph @dotnet/jit-contrib Given that this is a JIT32 issue, do we need to continue tracking it?

**Comment by RussKeldorph** [comment]
> That is the same issue exists even on RyuJIT based CoreCLR, when such a test is run against release build and generating debuggable code.

Does that mean, despite the bug title, there's still an issue for RyuJIT?  If so, I could be persuaded to change the bug title or close because there's little value in tracking it for RyuJIT.  I wouldn't close it because it's JIT32-specific if it's not.

**Comment by pgavlin** [comment]
> Does that mean, despite the bug title, there's still an issue for RyuJIT? If so, I could be persuaded to change the bug title or close because there's little value in tracking it for RyuJIT. I wouldn't close it because it's JIT32-specific if it's not.

Looks like it; we should update the title.

**Comment by RussKeldorph** [comment]
Changed title since apparently not JIT32-specific.  Not clear if x86 label is even appropriate.
